### ISSUE TITLE
[ioredis] Unhandled error event: Error: connect ECONNREFUSED

### ISSUE THREAD

**Issue by b4b857f6ee** [OpenCTI-Platform/opencti#1115]
Please replace every line in curly brackets { like this } with an appropriate answer, and remove this line.

## Description

Hello,

Information : i add information about redis to make it run on 6378 and change the docker-compose to match it.
on the docker opencti/platform:4.2.3 (he is always restarting) i have

[ioredis] Unhandled error event: Error: connect ECONNREFUSED 172.19.0.11:6378
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1129:14)
{"error":{"name":"UnknownError","data":{"message":"Connection is closed.","_stack":"Error: Connection is closed.\n    at Redi (events.js:300:26)\n    at Redis.emit (events.js:210:5)\n    at processTicksAndRejections (internal/process/task_queues.js:725T17:12:17.138Z"}

On the docker redis : 

1:signal-handler (1614270946) Received SIGTERM scheduling shutdown...
1:M 25 Feb 2021 16:35:47.027 # User requested shutdown...
1:M 25 Feb 2021 16:35:47.027 * Saving the final RDB snapshot before exiting.
1:M 25 Feb 2021 16:35:47.033 * DB saved on disk
1:M 25 Feb 2021 16:35:47.033 # Redis is now ready to exit, bye bye...
1:C 25 Feb 2021 16:36:39.772 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 25 Feb 2021 16:36:39.772 # Redis version=6.0.10, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 25 Feb 2021 16:36:39.772 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 25 Feb 2021 16:36:39.779 # Not listening to IPv6: unsupported
1:M 25 Feb 2021 16:36:39.782 * Running mode=standalone, port=6379.
1:M 25 Feb 2021 16:36:39.782 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 25 Feb 2021 16:36:39.782 # Server initialized
1:M 25 Feb 2021 16:36:39.782 * Loading RDB produced by version 6.0.10
1:M 25 Feb 2021 16:36:39.782 * RDB age 52 seconds
1:M 25 Feb 2021 16:36:39.782 * RDB memory usage when created 0.77 Mb
1:M 25 Feb 2021 16:36:39.783 * DB loaded from disk: 0.001 seconds
1:M 25 Feb 2021 16:36:39.783 * Ready to accept connections

on the docker opencti/worker

Traceback (most recent call last):
  File "worker.py", line 295, in <module>
    worker = Worker()
  File "worker.py", line 222, in __init__
    self.opencti_url, self.opencti_token, self.log_level
  File "/usr/local/lib/python3.7/site-packages/pycti/api/opencti_api_client.py", line 144, in __init__
    "OpenCTI API is not reachable. Waiting for OpenCTI API to start or check your configuration..."
ValueError: OpenCTI API is not reachable. Waiting for OpenCTI API to start or check your configuration...
INFO:root:Listing Threat-Actors with filters null.

i have no IPV6 allowing at the kernel level for information.

## Environment

1. OS Ubuntu 18.04
2. OpenCTI version: 4.2.3
3. OpenCTI client: front?
4. Other environment details: i have another redis docker on 6379, that's why i change the port

## Reproducible Steps

Steps to create the smallest reproducible scenario:
I just follow the procedure : https://www.notion.so/Using-Docker-03d5c0592b9d4547800cc9f4ff7be2b8

## Expected Output

Access to OpenCTI

## Actual Output

See the logs at the top of this message
 
## Additional information

No IPV6 allow at the kernel level + proxy needed to go to internet.

**Comment by richard-julien** [comment]
Hi,

Please check if you correctly share the network between your containers.

**Comment by b4b857f6ee** [comment]
> Hi,
> 
> Please check if you correctly share the network between your containers.

Ok i'm a noob in docker.

I just check this : https://docs.docker.com/compose/networking/

But i can't find any network part on your default docker-compose.yml

I miss 2 others information. I switch from 8080 to 9091
And change the url by something like toto.toto.com (my hosts file resolv it as the IP of my server) maybe it is the problem?

eq OPENCTI_URL=http://toto.toto.com:9091

**Comment by richard-julien** [comment]
We do not have any specific config in our docker compose because by definition all containers define inside the same compose will share their networks.

If you have deploy in multiple compose you need to define a network a add this network to your containers.

If you use server outside your docker containers you need to be sure that the hostname can be resolve and of course accessible

**Comment by b4b857f6ee** [comment]
ok so i create a network with docker, put the configuration at the end of the file.
use the default redis port
change the 8080 default (already use by docker) by 8081 for opencti.
All the docker run.
The local curl work, the local remote work (even if i have no data for now, i guess i have other configuration to do.)
But my access using cloudflare are not working. Did you ever try using it through cloudflare?
I got this :
![image](https://user-images.githubusercontent.com/36012515/109214749-81c53080-77b2-11eb-9e8b-333a12c9f15c.png)
![image](https://user-images.githubusercontent.com/36012515/109215092-f7310100-77b2-11eb-9d5d-ea025a7a164d.png)

**Comment by b4b857f6ee** [comment]
I think i still have issue with the connector now
opencti/connector-import-file-stix:4.2.3

OpenCTI API is not reachable. Waiting for OpenCTI API to start or check your configuration...
INFO:root:Listing Threat-Actors with filters null.

connector-export-file-csv:4.2.3

OpenCTI API is not reachable. Waiting for OpenCTI API to start or check your configuration...
INFO:root:Listing Threat-Actors with filters null.

connector-import-file-pdf-observables:4.2.3 same
connector-export-file-stix same

connector-history:4.2.3

Traceback (most recent call last):
  File "history.py", line 70, in <module>
    HistoryInstance = HistoryConnector()
  File "history.py", line 20, in __init__
    self.helper = OpenCTIConnectorHelper(config)
  File "/usr/local/lib/python3.8/site-packages/pycti/connector/opencti_connector_helper.py", line 309, in __init__
    self.api = OpenCTIApiClient(
  File "/usr/local/lib/python3.8/site-packages/pycti/api/opencti_api_client.py", line 143, in __init__
    raise ValueError(
ValueError: OpenCTI API is not reachable. Waiting for OpenCTI API to start or check your configuration...
INFO:root:Listing Threat-Actors with filters null.

**Comment by b4b857f6ee** [comment]
And i don't know where to configure the proxy, i already done this for MISP because we can configure this inside the misp configuration interface for proxy but i don't think they is something on opencti right?

**Comment by b4b857f6ee** [comment]
Ok so i rename all the hostname i change to the default one opencti:8081

now i got : 
- opencti/worker
INFO:root:Getting connectors ...
INFO:root:Getting connectors ...

- opencti/platform

{"error":{"name":"ConfigurationError","_error":{},"_showLocations":false,"_showPath":false,"time_thrown":"2021-02-25T20:56:29.792Z","data":{"reason":"ElasticSearch seems down","category":"technical"},"internalData":{}},"version":"4.2.3","level":"error","message":"[OPENCTI] Platform initialization fail","timestamp":"2021-02-25T20:56:29.792Z"}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
yarn run v1.19.1

- minio/minio
 You are running an older version of MinIO released 1 week ago
 Update: Run `mc admin update`


Endpoint: http://172.20.0.10:9000  http://127.0.0.1:9000

Browser Access:
   http://172.20.0.10:9000  http://127.0.0.1:9000

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide

- redis
1:M 25 Feb 2021 21:01:02.078 * 100 changes in 300 seconds. Saving...
1:M 25 Feb 2021 21:01:02.079 * Background saving started by pid 18
18:C 25 Feb 2021 21:01:02.083 * DB saved on disk
18:C 25 Feb 2021 21:01:02.084 * RDB: 0 MB of memory used by copy-on-write
1:M 25 Feb 2021 21:01:02.179 * Background saving terminated with success

- opencti/connector-export-file-stix

INFO:pika.adapters.utils.io_services_utils:Socket connected: <socket.socket fd=7, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('172.20.0.7', 51996), raddr=('172.20.0.4', 5672)>
INFO:pika.adapters.utils.connection_workflow:Streaming transport linked up: (<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f71d03b2730>, _StreamingProtocolShim: <SelectConnection PROTOCOL transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f71d03b2730> params=<ConnectionParameters host=rabbitmq port=5672 virtual_host=/ ssl=False>>).
INFO:pika.adapters.utils.connection_workflow:AMQPConnector - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f71d03b2730> params=<ConnectionParameters host=rabbitmq port=5672 virtual_host=/ ssl=False>>
INFO:pika.adapters.utils.connection_workflow:AMQPConnectionWorkflow - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f71d03b2730> params=<ConnectionParameters host=rabbitmq port=5672 virtual_host=/ ssl=False>>
INFO:pika.adapters.blocking_connection:Connection workflow succeeded: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7f71d03b2730> params=<ConnectionParameters host=rabbitmq port=5672 virtual_host=/ ssl=False>>
INFO:pika.adapters.blocking_connection:Created channel=1

- opencti/connector-export-file-csv same
- opencti/connector-import-file-stix same
- opencti/connector-history

INFO:root:HEARTBEAT:"2021-02-25T21:06:14.363Z"
INFO:root:HEARTBEAT:"2021-02-25T21:06:34.363Z"
INFO:root:HEARTBEAT:"2021-02-25T21:06:54.364Z"
INFO:root:HEARTBEAT:"2021-02-25T21:07:14.363Z"
INFO:root:HEARTBEAT:"2021-02-25T21:07:34.363Z"
INFO:root:HEARTBEAT:"2021-02-25T21:07:54.364Z"

I guess all looks good, i just need the proxy, i will try to find it in some issues maybe.

**Comment by b4b857f6ee** [comment]
Ok i guess i have to put it in all section of services : https://github.com/OpenCTI-Platform/opencti/issues/1036

**Comment by b4b857f6ee** [comment]
Ok now it's ok with the default name/proxy and network configure in every docker-compose file.
Just remain an misp issue but i'm looking into it my self and create a issues if i can't find why it is import 0.
But i still have the cloudflare issue.
I'm on 4.2.4

**Comment by SamuelHassine** [comment]
@b4b857f6ee  It seems that the CloudFlare issues is linked to CloudFlare JS not loading. Are you sure this is related to OpenCTI?

**Comment by b4b857f6ee** [comment]
> @b4b857f6ee It seems that the CloudFlare issues is linked to CloudFlare JS not loading. Are you sure this is related to OpenCTI?

Maybe not but, i have other application like MISP and i have no problem with cloudflare. Maybe someone already face it and know why :/.

**Comment by richard-julien** [comment]
![image](https://user-images.githubusercontent.com/285555/110352606-6a9a0480-8036-11eb-9b20-e2d95123997b.png)
![image](https://user-images.githubusercontent.com/285555/110352665-78e82080-8036-11eb-89ec-0591cb848352.png)
Looks like cloudflare blocked your request based on his security policy.
Do you have restriction on the referer or other security policies apply on your cloudflare account?

**Comment by b4b857f6ee** [comment]
> ![image](https://user-images.githubusercontent.com/285555/110352606-6a9a0480-8036-11eb-9b20-e2d95123997b.png)
> ![image](https://user-images.githubusercontent.com/285555/110352665-78e82080-8036-11eb-89ec-0591cb848352.png)
> Looks like cloudflare blocked your request based on his security policy.
> Do you have restriction on the referer or other security policies apply on your cloudflare account?

AHAH, i always missing tagging you on linkedin when something is about opencti aha. Sorry men!
Yes i have try remove all from my waf so it was a passthrough. But let me try again.

**Comment by SamuelHassine** [comment]
Hello,

I'm closing this issue due to inactivity. If the problem persists, please let us know!

Kind regards,
Samuel
